src/main/java/com/github/wovnio/wovnjava/WovnServletFilter.java

```diff
@@ -89,22 +89,22 @@ private void tryTranslate(Headers headers, RequestOptions requestOptions, HttpSe
             wovnRequest.getRequestDispatcher(headers.getCurrentContextUrlInDefaultLanguage().getPath()).forward(wovnRequest, wovnResponse);
         }
 
-        String originalBody = wovnResponse.toString();
-        if (originalBody != null) {
+        if (htmlChecker.isTextContentType(response.getContentType())) {
```

Comment:
Why is `htmlChecker.canTranslateContentType` not good enough? We only care about translating html anyway

Reply:
Or is there problems if text content goes through the binary path?

Reply:
A big reason why I implemented it separately is because I thought javascript should also be converted to UTF-8, because in my experience, there have been cases where javascript and HTML character codes don't match, which leads to a glitch.

Therefore, I thought that what can be translated and what should be converted to utf-8 are two different things, so I implemented the method separately.

Incidentally, wovnResponse.toString() is the return value of the implicitly converted utf-8 string.

Reply:
oh, so everything is already converted to UTF-8 anyway in `wovnResponse.toString()`. I think wovnjava isn't totally correct still. Maybe someone uses SHIFT-JIS for their CSS files which would be broken. This is an improvement though.

Reply:
Should CSS not be converted to utf-8?

Reply:
it doesn't necessarily have to be UTF-8. I don't think we support it yet though

Reply:
Ah, maybe it doesn't support rewriting @charset in CSS, so converting it to utf-8 would be a problem. You mean?

Reply:
Yes, you can use `@charset` to make it something other than utf-8. None of our backends would handle this currently though.

Reply:
I see, CSS should be left out of the UTF-8 conversion.

Reply:
I'm saying you should not worry about this now. If we use `wovnResponse.toString()` then we have to set utf-8 for all text responses. Ideally, we would use the same encoding that came from the original response instead of assuming utf-8

Reply:
Maybe I should change the name of the method, too, like `canUtf8ConvertableContentType()`?

Reply:
I think the current name is OK. `wovnResponse.toString()` doesn't actually convert non-utf8 to utf8 either

Reply:
Isn't this line converting to Utf-8?
https://github.com/WOVNio/wovnjava/blob/master/src/main/java/com/github/wovnio/wovnjava/WovnHttpServletResponse.java#L49

Reply:
If you look at how it is implemented, it is just doing `new String(data, "utf-8)` which is only interpreting the bytes as utf-8, not actually converting.

Reply:
Oh, I hadn't seen the implementation. If you do so, the character encoding specified by `OutputStreamWriter` will be forced to utf8, so it's better to get and set the character encoding of the original string, right?
https://github.com/WOVNio/wovnjava/blob/master/src/main/java/com/github/wovnio/wovnjava/WovnHttpServletResponse.java#L70

Reply:
No, I guess it's better to keep it that way.

Reply:
Something like that, ideally